internal/setup/setup.go

```diff
@@ -344,23 +351,36 @@ type yamlArchive struct {
 
 type yamlPackage struct {
 	Name      string               `yaml:"package"`
-	Archive   string               `yaml:"archive"`
-	Essential []string             `yaml:"essential"`
-	Slices    map[string]yamlSlice `yaml:"slices"`
+	Archive   string               `yaml:"archive,omitempty"`
+	Essential []string             `yaml:"essential,omitempty"`
+	Slices    map[string]yamlSlice `yaml:"slices,omitempty"`
 }
 
 type yamlPath struct {
-	Dir     bool    `yaml:"make"`
-	Mode    uint    `yaml:"mode"`
-	Copy    string  `yaml:"copy"`
-	Text    *string `yaml:"text"`
-	Symlink string  `yaml:"symlink"`
-	Mutable bool    `yaml:"mutable"`
+	Dir     bool    `yaml:"make,omitempty"`
+	Copy    string  `yaml:"copy,omitempty"`
+	Text    *string `yaml:"text,omitempty"`
+	Symlink string  `yaml:"symlink,omitempty"`
+	Mutable bool    `yaml:"mutable,omitempty"`
+
```

Comment:
Nit: Remove this extra newline.

Reply:
Yes, and let's please bring back the original order of fields on all types, as it was probably helping demonstrate semantics.  Right now Mode is next to Arch and Generate, and separated out in another block. Doesn't look right.